packages/preview-middleware-client/src/adp/controllers/AddFragment.controller.ts

```diff
@@ -0,0 +1,404 @@
+/** sap.m */
+import type Dialog from 'sap/m/Dialog';
+import MessageToast from 'sap/m/MessageToast';
+
+/** sap.ui.core */
+import { ValueState } from 'sap/ui/core/library';
+import type UI5Element from 'sap/ui/core/Element';
+import Controller from 'sap/ui/core/mvc/Controller';
+
+/** sap.ui.base */
+import type Event from 'sap/ui/base/Event';
+import type EventProvider from 'sap/ui/base/EventProvider';
+import type ManagedObject from 'sap/ui/base/ManagedObject';
+import type ManagedObjectMetadata from 'sap/ui/base/ManagedObjectMetadata';
+
+/** sap.ui.model */
+import JSONModel from 'sap/ui/model/json/JSONModel';
+
+/** sap.ui.rta */
+import type RuntimeAuthoring from 'sap/ui/rta/RuntimeAuthoring';
+
+/** sap.ui.dt */
+import OverlayRegistry from 'sap/ui/dt/OverlayRegistry';
+import type ElementOverlay from 'sap/ui/dt/ElementOverlay';
+
+import ControlUtils from '../control-utils';
+import CommandExecutor from '../command-executor';
+import { ManifestAppdescr, getFragments, getManifestAppdescr, writeFragment } from '../api-handler';
+
+type ExtendedEventProvider = EventProvider & {
```

Comment:
What is an `ExtendedEventProvider`? It feels like a type definition that should not exist here. Is this an event provider for a specific type of events? Should this move into the `types` section with a more meaningful name?

Reply:
Removed in: https://github.com/SAP/open-ux-tools/pull/1183/commits/07333f45e7c924ab2f3d32d4d86c57006fb1ffce